Source/WTF/wtf/glib/SysprofAnnotator.h

```diff
@@ -269,6 +269,15 @@ class SysprofAnnotator final {
         } else {
             unsigned newId = sysprof_collector_request_counters(1);
 
+            // Temporary workaround for libsysprof-capture providing conflicting IDs to threads.
+            static unsigned maxId = 0;
+            if (newId <= maxId) {
+                if (newId < maxId)
+                    sysprof_collector_request_counters(maxId - newId);
+                newId = sysprof_collector_request_counters(1);
+            }
```

Comment:
I think this can be simplifiied to:

```suggestion
            if (newId <= maxId)
                newId = sysprof_collector_request_counters(maxId - newId + 1);
```

No?

Reply:
Given:
```
sysprof_collector_request_counters (unsigned int n_counters)
{
  unsigned int ret = 0;

  if (n_counters == 0)
    return 0;

  COLLECTOR_BEGIN {
    ret = collector->next_counter_id;
    ((SysprofCollector *)collector)->next_counter_id += n_counters;
  } COLLECTOR_END;

  return ret;
}
```

The above wouldn't work as we'd obtain `next_counter_id` before it's increased. However, technically we could do:
```
newId = sysprof_collector_request_counters(maxId - newId + 1) + (maxId - newId) ;
```
what do you think?

Reply:
Maybe it needs a `+1` but that sounds good to me